### PR TITLE
[image_picker] fix crash on Android when selecting photo from OneDrive

### PR DESCRIPTION
**Pull request by shpp-vsmaga**
## Description
Fixed crash on Android when selecting photos from OneDrive. It stores files in the local storage, we can't get path and should work with it's URIs as remote, same like with GooglePhotos

## Checklist

Before you create this PR confirm that it meets all requirements listed below by checking the relevant checkboxes (`[x]`). This will ensure a smooth and quick review process.

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] My PR includes unit or integration tests for *all* changed/updated/fixed behaviors (See [Contributor Guide]).
- [x] All existing and new tests are passing.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] The analyzer (`flutter analyze`) does not report any problems on my PR.
- [x] I read and followed the [Flutter Style Guide].
- [x] The title of the PR starts with the name of the plugin surrounded by square brackets, e.g. [shared_preferences]
- [x] I updated pubspec.yaml with an appropriate new version according to the [pub versioning philosophy].
- [x] I updated CHANGELOG.md to add a description of the change.
- [x] I signed the [CLA].
- [x] I am willing to follow-up on review comments in a timely manner.

## Breaking Change

Does your PR require plugin users to manually update their apps to accommodate your change?

- [ ] Yes, this is a breaking change (please indicate a breaking change in CHANGELOG.md and increment major revision).
- [x] No, this is *not* a breaking change.

### CODE REVIEW

File: packages/image_picker/android/src/main/java/io/flutter/plugins/imagepicker/FileUtils.java [hunk 1]
```diff
@@ -215,4 +215,8 @@ private static boolean isMediaDocument(Uri uri) {
   private static boolean isGooglePhotosUri(Uri uri) {
     return "com.google.android.apps.photos.contentprovider".equals(uri.getAuthority());
   }
+
+  private static boolean isOneDriveUri(Uri uri) {
+    return "com.microsoft.skydrive.content.external".equals(uri.getAuthority());
```
Comment (cyanglaz) [inline packages/image_picker/android/src/main/java/io/flutter/plugins/imagepicker/FileUtils.java]: Is there a document I can check to see this is correct?

Comment (shpp-vsmaga) [reply packages/image_picker/android/src/main/java/io/flutter/plugins/imagepicker/FileUtils.java]: Unfortunately, I can't find it in any official documentation. Got it in an experimental way when selecting photos from OneDrive provider with uri.getAuthority()

**Comment by cyanglaz** [comment]
@shpp-vsmaga Unfortunately, I think it is risky to merge this in without a proper document or test. Or I wonder if we can find a safer way to solve this problem.

**Comment by cyanglaz** [comment]
I'm putting this to backlog since we don't have a proper document support for this change, nor do we have a proper tests.
If anyone can find a document somewhere for the scheme or we can find a way to test this, we can re-evaluate this PR.

**Comment by stuartmorgan** [comment]
Thank you for your contribution. I'm going to close this PR for now since there are outstanding comments, just to get this off our PR review queue. Please don't hesitate to submit a new PR if you have the time to address the review comments. Thanks!

(If you or someone else are interested in revisiting this, native unit tests are likely the easiest way to add testing here.)

**Pull request closed without merging**